File: python/arrays_and_hashing/1_two_sum.py
# https://leetcode.com/problems/two-sum/

class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        prev = {}
        for i in range(len(nums)):
            diff = target - nums[i]
            if diff in prev:
                return [i, prev[diff]]
            prev[nums[i]] = i

File: go/arrays_and_hashing/1_two_sum.go
package arraysandhashing

func TwoSum(nums []int, target int) []int {
	prev := make(map[int]int, len(nums))
	for i, n := range nums {
		diff := target - n
		if j, ok := prev[diff]; ok {
			return []int{i, j}
		}
		prev[n] = i
	}
	return []int{}
}
